oracle/src/pasta/mod.rs

```diff
@@ -2,3 +2,27 @@ pub mod fp_kimchi;
 pub mod fp_legacy;
 pub mod fq_kimchi;
 pub mod fq_legacy;
+
+use crate::poseidon::ArithmeticSpongeParams;
+use mina_curves::pasta::{Fp, Fq};
+use once_cell::sync::OnceCell;
+
+pub fn fp_kimchi_params() -> ArithmeticSpongeParams<Fp> {
```

Comment:
In general, I'm not a fan of bloating the library with memory leaks like these. IMO, it would be better to separate out the parameters from the constructor, and have a cachable `PoseidonConfig` that can be passed by reference to `Poseidon::new`, instead of the `new_legacy` and `new_kimchi` functions. Then, you can choose to cache the parameters -- or not -- in your use of the library.

It looks like you've done the reference side of that in #677 already, so it may make sense to start with that as the base for these changes.

Reply:
I think the same improvement is done separately by [another commit](https://github.com/o1-labs/proof-systems/commit/cb9a8877d4def4d8f22630a06d217c02a5c6c0fd#diff-857e792c5b38117d759c3d896c29ab7f3c2ffe06fd388b9e08f4cdb12ac46ae2R50) so I will close my 2 PRs.  

btw, the benchmark in the other PR https://github.com/o1-labs/proof-systems/pull/677 should still be valid and I think the `reset` functionality now creates non-trivial overhead, it's probably better to just remove the `reset`